stablehlo/dialect/StablehloOps.td

```diff
@@ -50,6 +57,13 @@ include "stablehlo/dialect/StablehloAttrs.td"
 class StableHLO_ShapedInterfaceOp<string mnemonic, list<Trait> traits> :
     StableHLO_Op<mnemonic, traits # [DeclareOpInterfaceMethods<InferShapedTypeOpInterface,
     ["reifyReturnTypeShapes"]>]> {
+  let extraClassDeclaration = [{
+    // Relax the strict default implementation with one that allows
+    // for StableHLO-specific differences.
+    static bool isCompatibleReturnTypes(TypeRange l, TypeRange r) {
+      return mlir::hlo::isCompatibleForHloTypeInference(l, r);
+    }
+  }];
```

Comment:
This extraClassDeclaration doesn't seem to be needed. Tests succeed without it.

Reply:
All tests are expect to succeed both before and after this PR: the intend of this PR is to loose the checks, so that in the future when we add dynamism in ops, like `ClampOp` which inherit this `StableHLO_ShapedInterfaceOp`, there will be not need to change the `.td` file any more: all the ops already are equipped a powerful check (inferred type VS real result type) by this PR.